api/v1alpha1/secretagentconfiguration_types.go

```diff
@@ -173,6 +173,7 @@ type AliasConfig struct {
 	CommonName     string          `json:"commonName,omitempty" yaml:"commonName,omitempty"`
 	Sans           []string        `json:"sans,omitempty" yaml:"sans,omitempty"`
 	SignedWithPath []string        `json:"signedWithPath,omitempty" yaml:"signedWithPath,omitempty"`
+	SharedCert     bool            `json:"sharedCert,omitempty" yaml:"sharedCert,omitempty"`
```

Comment:
So this bool is to generate an "invalid" cert used by DS. Not sure if "SharedCert" is a good name. Any ideas?

Reply:
I'm calling it shared, because it's the same cert mounted on all DS instances that's used to encrypt things between DS instances for sharing encrypted things... so?

Reply:
that's a ds specific use. I thought we wanted to make this generic.

Reply:
`placeholderCert`
`invalidCert`
`expiredCert`
`limitedCert`

IDK.. just rambling names that come to mind 😕

Reply:
I mean it is kind of. if we wanted to switch the bool, it would would be something like `selfsigned: true` `expire: 1971`. Its just a cert to encrypt shared things and shouldnt be used as PKI which is implied as usable for selfsigned. We can add those it just become more parameters that need to be passed around.  Maybe we could call it `EncryptionCert`

Reply:
`NonConformingCert`?

Reply:
If we want to make it 100% generic we can add `algorithm`,  `selfSigned`, and `expire` to `KeyConfig` . That would allow to get rid of the new function as well. Just need to pass these params to the  GenerateSignedCertPEM